### PR TITLE
swiftgen 5.3.0

### PR DESCRIPTION
**Pull request by AliSoftware**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)? [Kind of, see below]
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [x] Also passes `brew test swiftgen`

-----

Note: I tried to run `brew bump-formula-pr --strict swiftgen --tag="5.3.0" --revision="c2b03152f31e779f1deba3811b158ac1670f5b3b"`to create that PR as suggested in the updated `CONTRIBUTING.md` , but it failed when trying to run `rubocop`:
```
Already up-to-date.
==> replace "5.2.1" with "5.3.0"
==> replace "21107b1de6180cf5644e512e67e4d60a0d115ac4" with "c2b03152f31e779f1deba3811b158ac1670f5b3b"
Error: Error running `rubocop --format json --force-exclusion --parallel --except NewFormulaAudit --config /usr/local/Homebrew/Library/.rubocop_audit.yml /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/swiftgen.rb`
```
So I made the PR manually, the old way. The command `brew bump-formula-pr` wouldn't have been enough anyway (since the fixtures paths had to be updated), but might as well give you that feedback about the rubocop failure :wink:

### CODE REVIEW

**Comment by fxcoudert** [comment]
Fails on Sierra with:

```
Warning: Your Ruby version is compiled against a copy of OpenSSL that is very
old. Starting in January 2018, RubyGems.org will refuse connection requests from
these very old versions of OpenSSL. If you will need to continue installing gems
after January 2018, please follow this guide to upgrade:
http://ruby.to/tls-outdated.
Fetching gem metadata from https://rubygems.org/........
public_suffix-3.0.2 requires ruby version >= 2.1, which is incompatible with the
current version, ruby 2.0.0p648
```

Failed on High Sierra with:

```
bundler: failed to load command: xcpretty (/private/tmp/swiftgen-20180317-96086-18z1bu/gem_home/bin/xcpretty)
Errno::EPERM: Operation not permitted @ dir_s_mkdir - /xcode
```

File: Formula/swiftgen.rb [hunk 1]
```diff
@@ -2,8 +2,8 @@ class Swiftgen < Formula
   desc "Swift code generator for assets, storyboards, Localizable.strings, …"
   homepage "https://github.com/SwiftGen/SwiftGen"
   url "https://github.com/SwiftGen/SwiftGen.git",
-      :tag => "5.2.1",
-      :revision => "21107b1de6180cf5644e512e67e4d60a0d115ac4"
+      :tag => "5.3.0-2",
```
Comment (fxcoudert) [inline Formula/swiftgen.rb]: Github still lists 5.3.0 as latest release

File: Formula/swiftgen.rb [hunk 2]
```diff
@@ -22,21 +22,21 @@ def install
...
+      "Tests/Fixtures/Resources/XCAssets/Colors.xcassets" => "Colors.xcassets",
+      "Tests/Fixtures/Resources/Colors/colors.xml" => "colors.xml",
+      "Tests/Fixtures/Resources/Strings/Localizable.strings" => "Localizable.strings",
+      "Tests/Fixtures/Resources/Storyboards-iOS" => "Storyboards-iOS",
+      "Tests/Fixtures/Resources/Fonts" => "Fonts",
+      "Tests/Fixtures/Generated/XCAssets/swift3-context-all.swift" => "xcassets.swift",
+      "Tests/Fixtures/Generated/Colors/swift3-context-defaults.swift" => "colors.swift",
+      "Tests/Fixtures/Generated/Strings/structured-swift3-context-localizable.swift" => "strings.swift",
+      "Tests/Fixtures/Generated/Storyboards-iOS/swift3-context-all.swift" => "storyboards.swift",
+      "Tests/Fixtures/Generated/Fonts/swift3-context-defaults.swift" => "fonts.swift",
```
Comment (fxcoudert) [inline Formula/swiftgen.rb]: This sort of large diff of list of files to install shows why we need a more general install mechanism than is present currently. Maybe we could install in `pkgshare` all of `Tests/Fixtures`?

File: Formula/swiftgen.rb (hunk 1 above)
Comment (AliSoftware) [reply Formula/swiftgen.rb]: The tag 5.3.0-2 exists though https://github.com/SwiftGen/SwiftGen/releases/tag/5.3.0-2

As said in the commit message, that tag is temporary just to see if the changes made will properly fix that formula (which I'm not entirely convinced, especially about the error on High Sierra). Once we ensure we have found the right solution and the formula properly builds, I intend to delete tag 5.3.0-2 and create a real/official tag and squash the commits of this PR as if nothing bad happened.

File: Formula/swiftgen.rb (hunk 2 above)
Comment (AliSoftware) [reply Formula/swiftgen.rb]: All of Tests/Fixtures will be a lot, especially a lot of fixtures we have to tests storyboards, but also images (xcassets) & fonts. Those fixtures are not necessary at all for SwiftGen to work properly, and are only used by `brew test swiftgen`, nothing else, so seems like a shame to install all of `Tests/Fixtures` and unnecessary files and take space on the user's disk just for that, hence the cherry-picked list.

**Comment by AliSoftware** [comment]
@fxcoudert Any idea about the `Errno::EPERM: Operation not permitted @ dir_s_mkdir - /xcode` error from `xcpretty`? `xcpretty` worked for previous releases of SwiftGen and I don't remember updating it, so not sure what changed, maybe you updated some stuff on your CI machines since my last release?

**Comment by AliSoftware** [comment]
@fxcoudert Any info about the new build failures?

I found the logs on your Jenkins workers here: [Sierra](https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/version=sierra/20623/console) / [High Sierra](https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/version=high_sierra/20623/console), but couldn't find any exploitable info, they're probably hidden in the various `.log` files, which I don't think we have access to

**Comment by AliSoftware** [comment]
Seems that we still have the `xcpretty` error, and I don't get why, especially since it worked in previous releases of SwiftGen on homebrew. Did something change on your CI machines since last time we released some month ago?

**Comment by fxcoudert** [comment]
```
==> bundle exec rake cli:install[/usr/local/Cellar/swiftgen/5.3.0/bin,/usr/local/Cellar/swiftgen/5.3.0/lib,/usr/local/Cellar/swiftgen/5.3.0/share/swiftgen/templates]
== Building Binary ==
set -o pipefail && ( xcrun xcodebuild -workspace "SwiftGen.xcworkspace" -scheme "swiftgen" -configuration "Release" -derivedDataPath "/private/tmp/swiftgen-20180319-86573-1bbbdma/build" TEMPLATE_PATH="../share/swiftgen/templates" SWIFTGEN_OTHER_LDFLAGS="-sectcreate __TEXT __info_plist /private/tmp/swiftgen-20180319-86573-1bbbdma/build/Build/Products/Release/swiftgen.app/Contents/Info.plist") | tee "/cli_build_raw.log" | bundle exec xcpretty --color --report junit --output "/xcode/cli_build.xml"
tee: /cli_build_raw.log: Operation not permitted
```

It's using wrong absolute paths (`/cli_build_raw.log` and `/xcode/cli_build.xml`) so it seems like some environment variable not well defined (maybe those were `$prefix/cli_build_raw.log` and `prefix` is not begin defined as expected?)

**Comment by AliSoftware** [comment]
Ooooh I see. We wrote a test like `if ENV['CI']` but actually assumed at that time we were only gonna run on our own CI (CircleCI), forgetting about Homebrew's. Fixed by changing to `if ENV['CIRCLECI']`, hopefully it will fix the issue 🤞

**Comment by AliSoftware** [comment]
Now sure why this PR has been auto-closed by me, maybe I temporarily removed the branch. Anyway, continuing in #25475

**Pull request closed without merging**